inspire_utils/name.py

```diff
@@ -67,20 +67,66 @@ def __init__(self, name, constants=None):
         """Create a ParsedName instance.
 
         Args:
-            name (str): The name to be parsed (must be non empty nor None).
+            name (six.text_type): The name to be parsed (must be non empty nor None).
             constants (:class:`nameparser.config.Constants`): Configuration for `HumanName` instantiation.
                 (Can be None, if provided it overwrites the default one generated in
                 :method:`prepare_nameparser_constants`.)
         """
         if not constants:
             constants = ParsedName.constants
-        self.parsed_name = HumanName(name, constants=constants)
+
+        self._parsed_name = HumanName(name, constants=constants)
+
+        # In the case of one name part, i.e. only lastname, name parser adds the name part in one of the first, middle,
+        # suffix fields (depending on input). For our use-case if a user types a name, usually would be a lastname.
+        if len(name.split()) == 1:
+            self._parsed_name.last = self._parsed_name.first + self._parsed_name.middle + self._parsed_name.suffix
+            self._parsed_name.first = ''
+
+    def __len__(self):
+        return len(self._parsed_name)
 
     def __repr__(self):
-        return repr(self.parsed_name)
+        return repr(self._parsed_name)
 
     def __str__(self):
-        return str(self.parsed_name)
+        return str(self._parsed_name)
+
+    @property
+    def title(self):
+        return self._parsed_name.title
+
+    @property
+    def first(self):
+        return self._parsed_name.first
+
+    @property
+    def first_list(self):
+        return self._parsed_name.first_list
+
+    @property
+    def middle(self):
+        return self._parsed_name.middle
+
+    @property
+    def middle_list(self):
+        return self._parsed_name.middle_list
+
+    @property
+    def last(self):
+        return self._parsed_name.last
+
+    @property
+    def last_list(self):
+        return self._parsed_name.last_list
+
+    @property
+    def suffix(self):
+        return self._parsed_name.suffix
+
+    @property
+    def suffix_list(self):
+        return self._parsed_name.suffix_list
```

Comment:
It might be possible to avoid writing this long list of proxied properties with a little bit of `getattr` metaprogramming, but I'm ok with the current solution.

Reply:
Yes, I agree it would be better. I could do some refactoring in the future.